src/ipnet.rs

```diff
@@ -0,0 +1,828 @@
+    r#"([0-9a-fA-F]{1,4}:){1,7}:|"#,
+    r#"([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|"#,
+    r#"([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|"#,
+    r#"([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|"#,
+    r#"([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|"#,
+    r#"([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|"#,
+    r#"[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|"#,
+    r#":((:[0-9a-fA-F]{1,4}){1,7}|:)|"#,
+    r#"fe80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|"#,
+    r#"::(ffff(:0{1,4}){0,1}:){0,1}"#,
+    r#"((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}"#,
+    r#"(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])|"#,
+    r#"([0-9a-fA-F]{1,4}:){1,4}:"#,
+    r#"((25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])\.){3,3}"#,
+    r#"(25[0-5]|(2[0-4]|1{0,1}[0-9]){0,1}[0-9])"#,
+    r#")\/([0-9]|[1-9][0-9]|1[0-1][0-9]|12[0-8])$"#,
+);
+
+#[cfg(feature = "schemars")]
+impl schemars::JsonSchema for Ipv6Net {
+    fn schema_name() -> String {
+        "Ipv6Net".to_string()
+    }
+
+    fn json_schema(_: &mut schemars::gen::SchemaGenerator) -> schemars::schema::Schema {
+        schemars::schema::SchemaObject {
+            metadata: Some(Box::new(schemars::schema::Metadata {
+                title: Some("An IPv6 subnet".to_string()),
+                description: Some("An IPv6 subnet, including prefix and subnet mask".to_string()),
+                examples: vec!["fd12:3456::/64".into()],
+                ..Default::default()
+            })),
+            instance_type: Some(schemars::schema::InstanceType::String.into()),
+            string: Some(Box::new(schemars::schema::StringValidation {
+                pattern: Some(IPV6_NET_REGEX.to_string()),
+                ..Default::default()
+            })),
+            extensions: crate::schema_util::extension("Ipv6Net", "0.1.0"),
+            ..Default::default()
+        }
+        .into()
+    }
+}
+
+pub struct Ipv4NetIter {
+    next: Option<u32>,
+    last: u32,
+}
+
+impl Iterator for Ipv4NetIter {
+    type Item = Ipv4Addr;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let next = self.next?;
+        if next == self.last {
+            self.next = None;
+        } else {
+            self.next = Some(next + 1)
+        }
+        Some(next.into())
+    }
+
+    fn nth(&mut self, n: usize) -> Option<Self::Item> {
+        let next = self.next?;
+        let nth = next + n as u32;
+        (nth <= self.last).then_some(nth.into())
+    }
+}
+
+pub struct Ipv6NetIter {
+    next: Option<u128>,
+    last: u128,
+}
+
+impl Iterator for Ipv6NetIter {
+    type Item = Ipv6Addr;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let next = self.next?;
+        if next == self.last {
+            self.next = None;
+        } else {
+            self.next = Some(next + 1)
+        }
+        Some(next.into())
+    }
+
+    fn nth(&mut self, n: usize) -> Option<Self::Item> {
+        let next = self.next?;
+        let nth = next + n as u128;
```

Comment:
I think it's less likely since we're accepting `usize` in the signature, but this still seems like we should use `checked_add()` here, to avoid any possibility of panic.

Reply:
Totally